File: app/src/main/java/website/julianrosser/birthdays/activities/PrivacyPolicyActivity.java
package website.julianrosser.birthdays.activities;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v7.widget.Toolbar;
import android.widget.TextView;

import com.google.android.gms.analytics.GoogleAnalytics;
import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;

import website.julianrosser.birthdays.R;
import website.julianrosser.birthdays.Utils;

public class PrivacyPolicyActivity extends BaseActivity {

    private Tracker mTracker;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme();

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_privacy_policy);

        TextView textPrivacySummary = (TextView) findViewById(R.id.textPrivacySummary);
        setPrivacyTitleColour(textPrivacySummary);
        TextView textPrivacyFull = (TextView) findViewById(R.id.textPrivacyFull);
        setPrivacyTitleColour(textPrivacyFull);

        // Set up toolbar reference
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        // Show home button on toolbar
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        // Obtain the shared Tracker instance.
        mTracker = getDefaultTracker();

        mTracker.send(new HitBuilders.EventBuilder()
                .setCategory("Action")
                .setAction("Privacy Policy Activity")
                .build());
    }

    private void setPrivacyTitleColour(TextView textPrivacySummary) {
        int textColor = Utils.getHighlightColor(getApplicationContext());
        textPrivacySummary.setTextColor(getResources().getColor(textColor));
    }

    synchronized public Tracker getDefaultTracker() {
        if (mTracker == null) {
            GoogleAnalytics analytics = GoogleAnalytics.getInstance(this);
            mTracker = analytics.newTracker(R.xml.global_tracker);
        }
        return mTracker;
    }

    @Override
    protected void onResume() {
        super.onResume();
        mTracker.setScreenName("Privacy Policy");
        mTracker.send(new HitBuilders.ScreenViewBuilder().build());
    }

    // Set Activity theme depending on user preference
    public void setTheme() {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());

        if (prefs.getString(getResources().getString(R.string.pref_theme_key), "0").equals("0")) {
            setTheme(R.style.BlueTheme);
        } else if (prefs.getString(getResources().getString(R.string.pref_theme_key), "0").equals("1")) {
            setTheme(R.style.PinkTheme);
        } else if (prefs.getString(getResources().getString(R.string.pref_theme_key), "0").equals("2")) {
            setTheme(R.style.GreenTheme);
        } else {
            setTheme(R.style.PinkTheme);
        }
    }



}

File: app/src/main/java/website/julianrosser/birthdays/activities/PrivacyPolicyActivity.kt
package website.julianrosser.birthdays.activities

import android.os.Bundle
import android.preference.PreferenceManager
import android.widget.TextView
import com.google.android.gms.analytics.GoogleAnalytics
import com.google.android.gms.analytics.HitBuilders
import com.google.android.gms.analytics.Tracker
import kotlinx.android.synthetic.main.activity_privacy_policy.*
import kotlinx.android.synthetic.main.content_privacy_policy.*
import website.julianrosser.birthdays.R
import website.julianrosser.birthdays.Utils

class PrivacyPolicyActivity : BaseActivity() {

    private var mTracker: Tracker? = null

    private val defaultTracker: Tracker
        @Synchronized get() {
            if (mTracker == null) {
                val analytics = GoogleAnalytics.getInstance(this)
                mTracker = analytics.newTracker(R.xml.global_tracker)
            }
            return mTracker!!
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme()

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_privacy_policy)

        setPrivacyTitleColour(textPrivacySummary)
        setPrivacyTitleColour(textPrivacyFull)

        // Set up toolbar reference
        setSupportActionBar(toolbar)

        // Show home button on toolbar
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        // Obtain the shared Tracker instance.
        mTracker = defaultTracker

        mTracker?.send(HitBuilders.EventBuilder()
                .setCategory("Action")
                .setAction("Privacy Policy Activity")
                .build())
    }

    private fun setPrivacyTitleColour(textPrivacySummary: TextView) {
        val textColor = Utils.getHighlightColor(applicationContext)
        textPrivacySummary.setTextColor(resources.getColor(textColor))
    }

    override fun onResume() {
        super.onResume()
        mTracker?.let {
            it.setScreenName("Privacy Policy")
            it.send(HitBuilders.ScreenViewBuilder().build())
        }
    }

    // Set Activity theme depending on user preference
    private fun setTheme() {
        val prefs = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        setTheme(
                when (prefs.getString(resources.getString(R.string.pref_theme_key), "0")) {
                    "0" -> R.style.BlueTheme
                    "1" -> R.style.PinkTheme
                    "2" -> R.style.GreenTheme
                    else -> R.style.BlueTheme
                }
        )
    }

}
